var/spack/repos/builtin/packages/mrnet/package.py

```diff
@@ -3,16 +3,25 @@
 class Mrnet(Package):
     """The MRNet Multi-Cast Reduction Network."""
     homepage = "http://paradyn.org/mrnet"
-    url      = "ftp://ftp.cs.wisc.edu/paradyn/mrnet/mrnet_4.0.0.tar.gz"
+    url      = "ftp://ftp.cs.wisc.edu/paradyn/mrnet/mrnet_5.0.1.tar.gz"
+    list_url = "http://ftp.cs.wisc.edu/paradyn/mrnet"
 
-    version('4.0.0', 'd00301c078cba57ef68613be32ceea2f')
-    version('4.1.0', '5a248298b395b329e2371bf25366115c')
     version('5.0.1', '17f65738cf1b9f9b95647ff85f69ecdd')
+    version('4.1.0', '5a248298b395b329e2371bf25366115c')
+    version('4.0.0', 'd00301c078cba57ef68613be32ceea2f')
+
+    # Add a patch that brings mrnet-5.0.1 up to date with the current development tree
+    # The development tree contains fixes needed for the krell based tools
+    variant('krellpatch', default=False, description="Build MRNet with krell openspeedshop based patch.")
+    patch('krell-5.0.1.patch', when='@5.0.1+krellpatch')
```

Comment:
I have some additional krell related changes that aren't in the development tree yet.  Otherwise, yes that is a good idea.